@xen-orchestra/fs/src/s3.js

```diff
@@ -155,6 +155,13 @@ export default class S3Handler extends RemoteHandlerAbstract {
       if (e.name === 'EntityTooLarge') {
         return this._multipartCopy(oldPath, newPath)
       }
+      // normalize this error code
+      if (e.name === 'NoSuchKey') {
+        const error = new Error(`No such file  '${oldPath}`)
+        error.code = 'ENOENT'
```

Comment:
Keep the original error to ease debug:

```suggestion
        error.cause = e
        error.code = 'ENOENT'
```